### PR TITLE
blocklist implementation with genserver + tests

### PR DESCRIPTION
**Pull request by RobertJoonas**

### CODE REVIEW

File: lib/referrer_blocklist.ex [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    Process.cancel_timer(state[:timer])
+    new_timer = Process.send_after(self(), :update_list, @update_interval_milliseconds)
+
+    {:noreply, %{state | blocklist: updated_blocklist, timer: new_timer}}
+  end
+
+  defp get_blocklist(filepath, resource_url, opts \\ []) do
+    case HTTPoison.get(resource_url) do
+      {:ok, response} when response.status_code == 200 ->
+        if opts[:init], do: File.write!(filepath, response.body)
```
Comment (ukutaht) [inline lib/referrer_blocklist.ex]: What's the purpose of writing the response to the file?

**Review by ukutaht — CHANGES_REQUESTED** [review]
The code looks good. I think the logic could use some improvement.

The purpose of storing the spamlist in a file in the priv directory is two-fold:
1. To have a fallback if the request to github doesn't work
2. To be able to boot the app up quickly. Reading from a file is <1ms while an HTTP call might take more than a second.

The first objective is achieved with this PR but the second is not. The `init` function has to wait until the HTTP request is completed before it can serve `is_spammer?` requests. Ideally, the Genserver would be ready to respond to `is_spammer?` function calls immediately after reading the file, and the update would happen in the background.

To implement this, I think we should

File: lib/referrer_blocklist.ex (hunk 1 above)
Comment (RobertJoonas) [reply lib/referrer_blocklist.ex]: Deleted it now on the new commit.

Comment (RobertJoonas) [reply lib/referrer_blocklist.ex]: Deleted it now on the new commit.

**Comment by ukutaht** [comment]
Looks good!

**Pull request merged**